nntrainer/layers/concat_layer.cpp

```diff
@@ -27,11 +27,6 @@ ConcatLayer::ConcatLayer() : Layer(), leading_helper_dim(1) {}
 static constexpr size_t SINGLE_INOUT_IDX = 0;
 
 void ConcatLayer::finalize(InitLayerContext &context) {
-  if (context.getNumOutputs() != 1) {
```

Comment:
can concat layer support more than 1 output1?

Reply:
Now, this is governed by layer and layer node can query, so we shouldn't be querying num outputs here.

Reply:
but concat layer cannot support more than 1 output, and context.getNumOutputs()/context.getMinNumOutputs() must be there to check that error.
Then this will become `if (context.getNumOutputs() > 1), throw`.

Reply:
The mentioned check is done at
```cpp
initcontext::setOutputDimension() {
  NNTR_THROW_IF(out_dim.size() < num_requested_out, std::invalid_argument)
  ...
}
```

This has to be checked regardless of layer type, so factored out from each layer.

Reply:
sounds good.